setup.py

```diff
@@ -1,64 +1,119 @@
 from getpass import getpass
+from typing import Any, Union
 
 import requests
+from colorama import Fore, Style, init
 
 from core.utils.ip_utils import check_ip_address
 from core.wg.keygen import generate_private_key, generate_public_key
 
+init(autoreset=True)
 
 def make_config(path):
     # Create pair of crypto keys
     server_private_key = generate_private_key()
     server_public_key = generate_public_key(server_private_key)
-    ip_api = None
 
-    # Get external ip of your server
+    # [TelegramBot]
+    print(Style.BRIGHT + "--------- Bot Configuration ---------")
+    tg_token = getpass(Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Telegram bot token (hidden): ")
+    admins = input(Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Admin Telegram IDs with ',' separator: ")
+    faq_url = input_with_default(Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " FAQ URL (default: None): ", None)
+
+    # [db]
+    print(Style.BRIGHT + "--------- Database Configuration ---------")
+    db_path = input_with_default(Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Database path (default: db.sqlite): ", "db.sqlite")
+
+    # [core]
+    print(Style.BRIGHT + "--------- Core Configuration ---------")
+    peer_active_time = input_with_default(
+        Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Peer active time in hours (default: 12): ", 12
+    )
+    connection_listen_timer = input_with_default(
+        Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Connection listen timer in seconds (default: 120): ", 120
+    )
+    connection_update_timer = input_with_default(
+        Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Connection update timer in seconds (default: 300): ", 300
+    )
+    connection_connected_only_listen_timer = input_with_default(
+        Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Connection connected only listen timer in seconds (default: 60): ", 60
+    )
+    logs_path = input_with_default(Fore.YELLOW + "[ ! ]" + Style.RESET_ALL + " Logs path (default: logs): ", "logs")
+    os.makedirs(logs_path, exist_ok=True)
+    os.makedirs(os.path.join(logs_path, "bot"), exist_ok=True)
+    os.makedirs(os.path.join(logs_path, "core"), exist_ok=True)
+    print(Fore.GREEN + "[ + ]" + Style.RESET_ALL + " Logs path created")
+
+    # [Server]
+    print(Style.BRIGHT + "--------- Server Configuration ---------")
     try:
         ip_api = requests.get('https://api.ipify.org', timeout=5)
         if ip_api.status_code != 200:
             raise ValueError
         external_ip = ip_api.content.decode("utf-8")
+        print(Fore.GREEN + "[ + ]" + Style.RESET_ALL + " External IP fetched successfully")
     except (requests.exceptions.Timeout, ValueError):
-        print("ipify is not accessible!")
-        external_ip = input("[ ! ] Enter external ip manually: ")
-    tg_token = getpass("[ ! ] Telegram bot token: ")
-    admins = input("[ ! ] Admin Telegram IDs with ',' separator: ")
+        print(Fore.RED + "ipify is not accessible!")
```

Comment:
missing RESET_ALL, unlikely to matter here though

Reply:
it's intended